src/main/java/org/kiwiproject/dropwizard/poller/ClientPoller.java

```diff
@@ -0,0 +1,492 @@
+        }
+
+        checkState(nonNull(supplierTimeout), "supplierTimeout cannot be null");
+        checkState(supplierTimeout > 0, "supplierTimeout must be greater than zero");
+        checkState(nonNull(supplierTimeoutUnit), "supplierTimeoutUnit cannot be null");
+        checkState(nonNull(initialExecutionDelay), "initialExecutionDelay cannot be null");
+        checkState(nonNull(consumer), "consumer cannot be null");
+        checkState(nonNull(decisionFunction), "decisionFunction cannot be null");
+        checkState(executionInterval > 0, "executionInterval must be a positive number of milliseconds!");
+        checkState(nonNull(executor), "executor cannot be null");
+        checkState(nonNull(consumerExecutor), "consumerExecutor cannot be null");
+    }
+
+    private void poll() {
+        try {
+            if (shouldNotExecutePoll()) {
+                LOG.trace("{} - Skipping poll; decision function returned false", name);
+                statistics().incrementSkipCount();
+                return;
+            }
+            executePoll();
+        } catch (Exception ex) {
+            LOG.error("{} - Encountered unexpected error during polling: {}", name, ex.getMessage(), ex);
+        }
+    }
+
+    private boolean shouldNotExecutePoll() {
+        return !shouldExecutePoll();
+    }
+
+    private boolean shouldExecutePoll() {
+        try {
+            return decisionFunction.apply(statistics());
+        } catch (Exception ex) {
+            LOG.warn("{} - Encountered an exception while making polling decision, returning FALSE - {}", name, ex.getMessage());
+            LOG.debug("{} - Decision function exception details", name, ex);
+            return false;
+        }
+    }
+
+    private void executePoll() {
+        LOG.trace("{} - Poller executing", name);
+        long start = System.currentTimeMillis();
+        Response response = executePollRequest();
+        handleResponse(response);
+        long elapsed = System.currentTimeMillis() - start;
+        statistics().addPollLatencyMeasurement(elapsed);
+        LOG.trace("{} - Poll time: {} millis", name, elapsed);
+    }
+
+    @SuppressWarnings({"squid:51612", "Convert2MethodRef"})
+    private Response executePollRequest() {
+        statistics().incrementCount();
+        try {
+            SyncInvoker invoker = supplier.get();
+            updateUri(invoker);
+            return withMaxTimeout(doAsync(() -> invoker.get()), supplierTimeout, supplierTimeoutUnit).get();
+        } catch (final Exception e) {
+            statistics().incrementFailureCount(e);
+            LOG.error("{} - Poller HTTP GET request failed. URI: {}", name, uriOrDefault());
+            LOG.info("{} - Poller failure:", name, e);
+            return null;
+        } finally {
+            uri = null;
+        }
+    }
+
+    private void updateUri(SyncInvoker invoker) {
```

Comment:
Curious why we/I/you are _always_ calling this method on every poll (see above in `executePollRequest`). It should not change should it? Not suggesting to change it right now, but if you don't remember exactly why we did this, then maybe we should add an "investigation" task... 🤔

Reply:
I don't remember why this was done on every call